src/main/java/org/rcsb/rcsbsequencecoordinates/collectors/alignments/SequenceAlignmentsCollector.java

```diff
@@ -2,40 +2,48 @@
  *  Copyright (c) 2024 - now, RCSB PDB and contributors, licensed under MIT, See LICENSE file for more info.
  */
 
-package org.rcsb.collectors.alignments;
+package org.rcsb.rcsbsequencecoordinates.collectors.alignments;
 
 import org.bson.Document;
 import org.bson.conversions.Bson;
-import org.rcsb.collectors.map.MapCollector;
+import org.rcsb.rcsbsequencecoordinates.collectors.map.MapCollector;
 import org.rcsb.graphqlschema.reference.GroupReference;
 import org.rcsb.graphqlschema.reference.SequenceReference;
-import org.rcsb.utils.MongoStream;
+import org.rcsb.rcsbsequencecoordinates.utils.ReactiveMongoResource;
 import org.springframework.data.mongodb.core.index.IndexDirection;
+import org.springframework.stereotype.Service;
 import reactor.core.publisher.Flux;
 
 import java.util.List;
 
 import static com.mongodb.client.model.Aggregates.match;
 import static com.mongodb.client.model.Filters.*;
-import static org.rcsb.collectors.alignments.AlignmentsMongoHelper.*;
-import static org.rcsb.collectors.alignments.AlignmentsReferenceHelper.equivalentReferences;
-import static org.rcsb.collectors.alignments.AlignmentsReferenceHelper.testGenome;
+import static org.rcsb.rcsbsequencecoordinates.collectors.alignments.AlignmentsMongoHelper.*;
+import static org.rcsb.rcsbsequencecoordinates.collectors.alignments.AlignmentsReferenceHelper.equivalentReferences;
+import static org.rcsb.rcsbsequencecoordinates.collectors.alignments.AlignmentsReferenceHelper.testGenome;
 
 /**
  * @author : joan
- * @mailto : joan.segura@rcsb.org
- * @created : 2/23/24, Friday
- **/
+ */
+@Service
 public class SequenceAlignmentsCollector implements AlignmentsCollector {
 
-    public static AlignmentsCollector request(String queryId, SequenceReference from, SequenceReference to) {
+    private final ReactiveMongoResource mongoResource;
+    private final MapCollector mapCollector;
+
+    public SequenceAlignmentsCollector(ReactiveMongoResource mongoResource) {
+        this.mongoResource = mongoResource;
+        this.mapCollector = new MapCollector(mongoResource);
+    }
+
+    public AlignmentsCollector request(String queryId, SequenceReference from, SequenceReference to) {
         if(testGenome(from, to))
-            return GenomeAlignmentsCollector.request(queryId, from, to);
-        return ProteinAlignmentsCollector.request(queryId, from, to);
+            return GenomeAlignmentsCollector.request(queryId, from, to, mongoResource, this);
```

Comment:
This doesn't look good. Can you add a TODO so I remember to remove the circular dependency? My first impression is that `mapIds` does not belong here.

Reply:
Agreed, I didn't like this either. I feel like to solve this properly we'll need quite some restructuring of the collectors. I'll add a TODO.